### PR TITLE
Namaless patch 1

### PR DESCRIPTION
**Pull request by namaless**

### CODE REVIEW

File: src/Twig.php [hunk 1]
```diff
@@ -6,10 +7,12 @@
  * @copyright Copyright (c) 2011-2015 Josh Lockhart
  * @license   https://github.com/slimphp/Twig-View/blob/master/LICENSE.md (MIT License)
  */
+
 namespace Slim\Views;
 
+use Pimple\Container;
+use Pimple\ServiceProviderInterface;
```
Comment (akrabat) [inline src/Twig.php]: This component deliberately doesn't couple to Pimple so that it can be used with other DICs.

**Comment by akrabat** [comment]
Can you provide an explanation of why you wrote this PR please? e.g. what problem this PR solves or how it enables easier development, or whatever.

File: src/Twig.php (hunk 1 above)
Comment (namaless) [inline src/Twig.php]: I removed the fullname and added the use statment on top.

File: src/Twig.php [hunk 2]
```diff
@@ -77,14 +80,13 @@ public function register(\Pimple\Container $container)
     /**
      * Proxy method to add an extension to the Twig environment
      *
-     * @param array|object $extension A single extension instance or an array of instances
+     * @param \Twig_ExtensionInterface $extension A single extension instance or an array of instances
```
Comment (namaless) [inline src/Twig.php]: Use Twig_ExtensionInterface so not possible to pass array or other object.

File: src/Twig.php [hunk 3]
```diff
@@ -132,7 +134,7 @@ public function getLoader()
     /**
      * Return Twig environment
      *
-     * @return \Twig_EnvironmentInterface
+     * @return \Twig_Environment
```
Comment (namaless) [inline src/Twig.php]: Twig_EnvironmentInterface not exist.

File: src/TwigExtension.php [hunk 4]
```diff
@@ -6,48 +7,74 @@
...
     {
         return $this->router->pathFor($name, $data, $queryParams);
     }
 
-    public function baseUrl()
+    /**
+     * @param string $path
+     * @return string
+     */
+    public function baseUrl($path = '')
```
Comment (namaless) [inline src/TwigExtension.php]: Give the possible to use the base_url twig function with path, example:

{{ base_url('/assets/css/style.css') }}

instead

{{ base_url() }}/assets/css/style.css

Btw the old code working fine without any other modification.

**Comment by namaless** [comment]
Sorry i posted fast because i working on my new project, i think is easy to understand but i post some comments :)

**Comment by piotr-cz** [comment]
So this pull request add typecasts in method parameters and uses shorthand syntax for arrays as minimum requirement for package is PHP 5.4.

**Comment by akrabat** [comment]
Needs to be rebased and the Pimple stuff removed.

**Pull request closed without merging**